### DISCUSSION TITLE
Whether there is a way to use the model's type?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [prisma/prisma#20274] [Q&A] [answered]
in prisma/schema.prisma
```
model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
  published Boolean @default(false)
  authorId  Int?
  User      User?   @relation(fields: [authorId], references: [id])
}
```

my project is typescript.
whether there is a way to use the model's type? is there a existing approach to do it?

**Comment by ludralph** [comment, marked as answer]
Hi @liaodalin19903 👋 

You can use a model's generated type in Prisma. See the documentation on [Advanced Type Safety](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety)

You should be able to import any of Prisma's generated types from the @prisma/client package.

```prisma
model Post {
 ...
}
```
```js
import type { Post } from '@prisma/client'
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
